tests/main/snaps-state/task.yaml

```diff
@@ -0,0 +1,80 @@
+summary: smoke test for the snaps-state tool
+
+prepare: |
+    snap set system experimental.parallel-instances=true
+
+restore: |
+    snap set system experimental.parallel-instances=null
+
+execute: |
+    SNAP_NAME=test-snapd-tools
+    SNAP_CLASSIC=test-snapd-classic-confinement
+    SNAP_DEVMODE=test-snapd-devmode
+    SNAP_JAILMODE=test-devmode-cgroup
+
+    # Check help
+    "$TESTSTOOLS"/snaps-state | MATCH "usage: pack-local <snap-name>"
+    "$TESTSTOOLS"/snaps-state -h | MATCH "usage: pack-local <snap-name>"
+    "$TESTSTOOLS"/snaps-state --help | MATCH "usage: pack-local <snap-name>"
+
+    # Pack a local snap by using the pack-local subcommand
+    snap_path=$("$TESTSTOOLS"/snaps-state pack-local "$SNAP_NAME")
+    snap install --dangerous "${snap_path}"
+    test-snapd-tools.echo test123 | MATCH "test123"
+    snap remove "$SNAP_NAME"
+
```

Comment:
Can you please add a test that `packs` something which doesn't exist to see how it behaves please?

Reply:
done